handler/todo_handler.go

```diff
@@ -7,23 +7,40 @@ import (
 	"go-todo-api/utils"
 	"net/http"
 	"strconv"
+	"strings"
 
 	"github.com/gin-gonic/gin"
 	"go.uber.org/zap"
 )
 
 type ToDoHandler struct {
-	ToDoRepository repository.ToDoRepositoryInterface
+	ToDoRepository  repository.ToDoRepositoryInterface
+	RedisRepository repository.RedisClientInterface
 }
 
-func NewToDoHandler(ToDoRepo repository.ToDoRepositoryInterface) *ToDoHandler {
-	return &ToDoHandler{ToDoRepository: ToDoRepo}
+func NewToDoHandler(ToDoRepo repository.ToDoRepositoryInterface, redisRepository repository.RedisClientInterface) *ToDoHandler {
+	return &ToDoHandler{ToDoRepository: ToDoRepo, RedisRepository: redisRepository}
 }
 
 func (handler *ToDoHandler) PostToDo(c *gin.Context) {
-	var createToDo = &models.ToDoRequest{}
+	userID, err := strconv.ParseUint(c.Param("user_id"), 10, 64)
+	if err != nil {
+		zap.S().Error("Error: ", zap.Error(err))
+		c.JSON(http.StatusBadRequest, nil)
+		return
+	}
 
-	if err := c.BindJSON(&createToDo); err != nil {
+	redisToken := handler.RedisRepository.GetData(c.Param("user_id"))
```

Comment:
e peki token valid mi

Reply:
valid olmayan tokenler Rediste olmadığı için eğer token varsa validtir dedim

Reply:
hatta Authorization.go da isTokenValid var ama kullanmadım